IFComp/lib/IFComp/Controller/Admin/Email.pm

```diff
@@ -0,0 +1,50 @@
+package IFComp::Controller::Admin::Email;
+use Moose;
+use namespace::autoclean;
+
+use List::Util qw(none);
+
+BEGIN { extends 'Catalyst::Controller'; }
+
+sub index : Chained("/admin/root") : PathPart('email') : Args(0) {
+    my ( $self, $c, ) = @_;
+
+    my $comp = $c->model('IFCompDB::Comp')->current_comp;
+
+    my @emails = _get_emails( $c, $comp, 0 );
+
+    my @disqualified_emails = _get_emails( $c, $comp, 1 );
+
+    # The "anti-email" list is all holders of disqualified games that
+    # don't also hold qualified games.
+    my @anti_emails = grep {
+        my $dq_email = $_;
+        none { $_ eq $dq_email } @emails;
+    } @disqualified_emails;
+
+    $c->stash(
+        emails      => \@emails,
+        anti_emails => \@anti_emails,
+    );
+
+}
+
+sub _get_emails {
```

Comment:
Please move this method into the schema, along with the calculation of the anti_emails above. Ideally the controller will call a method (or two) on some result/resultset to retrieve the two lists, and then just put the lists into the stash for the template. The controller shouldn't have to do any direct query writing or list manipulation.